BrainPortal/app/models/work_group.rb

```diff
@@ -77,20 +77,28 @@ def self.prepare_pretty_category_names(groups = [], as_user = nil)
     wgs
   end
 
-  def pretty_category_name(as_user) #:nodoc:
+  def pretty_category_name(as_user = nil) #:nodoc:
     return @_pretty_category_name if @_pretty_category_name
-    if self.invisible?
-      @_pretty_category_name = 'Invisible Project'
-    elsif self.public?
-      @_pretty_category_name = 'Public Project'
-    elsif self.users.count == 0
-      @_pretty_category_name = 'Empty Work Project'
-    elsif self.users.count > 1
-      @_pretty_category_name = 'Shared Work Project'
-    elsif as_user.present? && (self.creator_id == as_user.id || self.users.first.id == as_user.id)
-      @_pretty_category_name = 'My Work Project'
+    if @_pretty_category_name.blank?
+      if self.invisible?
+        @_pretty_category_name = 'Invisible Project'
+      elsif self.public?
+        @_pretty_category_name = 'Public Project'
+      elsif self.users.count == 0
+        @_pretty_category_name = 'Empty Work Project'
+      elsif self.users.count > 1
+        @_pretty_category_name = 'Shared Work Project'
+      elsif as_user.present? && (self.creator_id == as_user.id || self.users.first.id == as_user.id)
+        @_pretty_category_name = 'My Work Project'
+      else
+        @_pretty_category_name = "Personal Work Project of #{self.users.first.full_name}"
+      end
+    end
+    # Public is more of quantifier, there could be public shared or public personal projects
```

Comment:
Please adjust the code between line 97 and 102

1. The `@_pretty_category_name.present?` is always true
2. There is no need for a `else` !!!!

Reply:
Also, based on the logic, you will never re-sinsert `" Public "` on anything but `Invisible Project`. You'll never get for instance 'Share Public Work Project' because of the if/elif structure above.

Reply:
Can I just suggest you set up, early, a string variable that has either `""` or `"Public "` and then you just insert it in the original category names, instead of trying to retrofit the word at the end?

Reply:
hi so would you like 

a) adjust code to enable Share Public Work Project or
b) just simplify new lines a bit
c) return old code

at this moment I will be fine with c) to avoid any perceived blame for the existing production bug shown on snapshot ( I factored it out into a separate issue https://github.com/aces/cbrain/issues/1323 we can fix it once agree on the best way to do it)

Reply:
OK.

I think honestly that the original code required very little changes. It wasn't buggy.

If you the filter list not behaving the way you expect, the problem is not in this method. Like I said, look for other places where the instance variable is modified.